lib/ViewModels/DataCatalogTabViewModel.js

```diff
@@ -38,4 +40,32 @@ DataCatalogTabViewModel.prototype.showInfo = function(item) {
     CatalogItemInfoViewModel.open('ui', item);
 };
 
+DataCatalogTabViewModel.prototype.clickEnabled = function(item) {
+    ga('send', 'event', 'dataSource', 'clickItem', item.name);
+
+    if (!item.isEnabled && defined(item.initialMessage) && item.initialMessage.length > 0) {
+        PopupMessageViewModel.open('ui', {
+            title: 'Data Set Information',
+            message: '<div>'+ item.initialMessage +'</div>'
+        });
+        item.initialMessage = undefined;
+    }
+
+    item.toggleEnabled();
+};
+
+DataCatalogTabViewModel.prototype.clickOpen = function(item) {
+    ga('send', 'event', 'dataSource', 'clickOpen', item.name);
+
+    if (!item.isOpen && defined(item.initialMessage) && item.initialMessage.length > 0) {
+        PopupMessageViewModel.open('ui', {
+            title: 'Data Provider Information',
+            message: '<div>'+ item.initialMessage +'</div>'
+        });
+        item.initialMessage = undefined;
```

Comment:
I don't like the lossiness of this.  For one thing, if you enable a catalog item, then use the Tools menu to download the catalog as JSON, the JSON won't contain the `initialMessage`.  A separate property to track whether its already shown would be better.  The property should be made non-serialized by adding a do-nothing handler for it to `defaultSerializers`.